Install libcpp from release package other than apt-get

cccclai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #11832



A few changes

1. Use the release package from https://releases.llvm.org/ to install libcpp, so we can install from different linux in addition to Ubuntu
2. Seperate out the qnn version and the url to a seperate file, for better versin management in the future

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745/)

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745)

cccclai committed:
Install libcpp from release package other than apt-get



Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745/)

[ghstack-poisoned]

cccclai committed:
Update on "Install libcpp from release package other than apt-get"




A few changes

1. Use the release package from https://releases.llvm.org/ to install libcpp, so we can install from different linux in addition to Ubuntu
2. Seperate out the qnn version and the url to a seperate file, for better versin management in the future

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745/)

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745)

[ghstack-poisoned]

cccclai committed:
Update on "Install libcpp from release package other than apt-get"




A few changes

1. Use the release package from https://releases.llvm.org/ to install libcpp, so we can install from different linux in addition to Ubuntu
2. Seperate out the qnn version and the url to a seperate file, for better versin management in the future

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745/)

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745)

[ghstack-poisoned]

cccclai committed:
Update on "Install libcpp from release package other than apt-get"




A few changes

1. Use the release package from https://releases.llvm.org/ to install libcpp, so we can install from different linux in addition to Ubuntu
2. Seperate out the qnn version and the url to a seperate file, for better versin management in the future

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745/)

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745)

[ghstack-poisoned]

cccclai committed:
Update on "Install libcpp from release package other than apt-get"




A few changes

1. Use the release package from https://releases.llvm.org/ to install libcpp, so we can install from different linux in addition to Ubuntu
2. Seperate out the qnn version and the url to a seperate file, for better versin management in the future

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745/)

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745)

[ghstack-poisoned]

shewu-quic commented on backends/qualcomm/scripts/install_qnn_sdk.sh:
```diff
@@ -0,0 +1,137 @@
+  echo "Set QNN_SDK_ROOT=${QNN_SDK_ROOT}"
+}
+
+setup_libcpp() {
```
Would you like to add a check if libcpp is installed or not?

cccclai committed:
Update on "Install libcpp from release package other than apt-get"




A few changes

1. Use the release package from https://releases.llvm.org/ to install libcpp, so we can install from different linux in addition to Ubuntu
2. Seperate out the qnn version and the url to a seperate file, for better versin management in the future

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745/)

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745)

[ghstack-poisoned]

cccclai commented:
@shewu-quic does it look good now?

shewu-quic commented on backends/qualcomm/scripts/install_qnn_sdk.sh:
```diff
@@ -0,0 +1,145 @@
+        return
+    fi
+
+    NDK_VERSION="r25c"
```
I think it can align with document with r26c.

shewu-quic commented on backends/qualcomm/scripts/install_qnn_sdk.sh:
```diff
@@ -0,0 +1,145 @@
+
+setup_libcpp() {
+  clang_version=$1
+  LLVM_VERSION="12.0.0"
```
Based on Qnn Document, it recommends using clang++14 .
<img width="1140" height="229" alt="image" src="https://github.com/user-attachments/assets/f5d3f01c-d6eb-4c02-afcd-c3d10913ed66" />

shewu-quic commented on backends/qualcomm/scripts/install_qnn_sdk.sh:
```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
```
Would you like to change the script permission such as 755?

shewu-quic replied:
Woops, Sorry for the confusion—I meant this script, not this line.

cccclai replied:
I see, does it mean I need to add instructions in readme to change the permission for this script? Just trying to understand your intention better

shewu-quic replied:
Yes, I think it would be better for general user :)

shewu-quic replied:
Or I think you can change this file permission and push again.
https://stackoverflow.com/questions/10516201/updating-and-committing-only-a-files-permissions-using-git-version-control
https://stackoverflow.com/questions/59160266/git-and-permissions/59163100#59163100

cccclai replied:
Actually maybe let me try not requiring the permission

shewu-quic reviewed (approved):
Thanks your effort.
LGTM.

cccclai committed:
Update on "Install libcpp from release package other than apt-get"




A few changes

1. Use the release package from https://releases.llvm.org/ to install libcpp, so we can install from different linux in addition to Ubuntu
2. Seperate out the qnn version and the url to a seperate file, for better versin management in the future

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745/)

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745)

[ghstack-poisoned]

cccclai commented:
@shewu-quic thanks for reviewing! Addressed the comments.

cccclai committed:
Update on "Install libcpp from release package other than apt-get"




A few changes

1. Use the release package from https://releases.llvm.org/ to install libcpp, so we can install from different linux in addition to Ubuntu
2. Seperate out the qnn version and the url to a seperate file, for better versin management in the future

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745/)

Differential Revision: [D77048745](https://our.internmc.facebook.com/intern/diff/D77048745)

[ghstack-poisoned]

shewu-quic reviewed (approved):
Thanks. LGTM.
